src/renderer/domains/network/registry/service/chainRegistry.ts

```diff
@@ -0,0 +1,125 @@
+import { type PolkadotClient, createClient } from 'polkadot-api';
+import { WsEvent, type WsJsonRpcProvider, getWsProvider } from 'polkadot-api/ws-provider/web';
+
+import { chainsService } from '@/shared/api/network';
+import { type Chain, type ChainId } from '@/shared/core';
+import { nullable } from '@/shared/lib/utils';
+import { CONFIG } from '../lib/constants';
+import { type ChainApi } from '../lib/types';
+
+class ChainRegistry {
+  // TODO: support Light Clients in future
+  #storage: Map<
+    ChainId,
+    {
+      provider: WsJsonRpcProvider;
+      client: PolkadotClient;
+    }
+  > = new Map();
+
+  #chainsList: Chain[];
+  #chainsMap: Map<ChainId, Chain>;
+
+  constructor(chains: Chain[]) {
+    this.#chainsList = chains;
+    this.#chainsMap = new Map(chains.map(c => [c.chainId, c]));
+  }
+
+  getApi(chainId: ChainId): ChainApi {
+    const connector = this.#storage.get(chainId);
+
+    if (nullable(connector)) {
+      throw new Error(`Provider and Client for ${chainId} is absent, need to establish connect first`);
+    }
+
+    if (nullable(CONFIG[chainId])) {
+      throw new Error(`Chain ${chainId} is not supported`);
+    }
+
+    return CONFIG[chainId](connector.client);
+  }
+
+  connect(
+    chainId: ChainId,
+    endpoints: string[],
+    handlers?: {
```

Comment:
Events should be generalized, each event should pass chain id as data in handler
```
const events = mitt<{}>()

function connect(chainId) {
  const provider = createProvider({
    onStatusChange(status) {
      events.emit(status, chainId)
    }
  })
}
```